apps/src/lib/util/AnalyticsReporter.js

```diff
@@ -0,0 +1,134 @@
+import {
+  init,
+  track,
+  Identify,
+  identify
+  // setUserId
+} from '@amplitude/analytics-browser';
+import {currentLocation} from '@cdo/apps/utils';
+
+// A flag that can be toggled to send events regardless of environment
+const ALWAYS_PUT = false;
+
+const Environments = {
+  production: 'production',
+  levelbuilder: 'levelbuilder',
+  test: 'test',
+  staging: 'staging',
+  adhoc: 'adhoc',
+  development: 'development',
+  unknown: 'unknown'
+};
+
+class AnalyticsReporter {
+  constructor() {
+    // Get the API key from the DOM. See application.html.haml layout.
+    const element = document.querySelector('script[data-amplitude-api-key]');
+    const api_key = element ? element.dataset.amplitudeApiKey : '';
+    // Let Amplitude handle missing/invalid keys.
+    if (this.shouldPutRecord(ALWAYS_PUT)) {
+      init(api_key);
+    }
+  }
+
+  setUserProperties(userId, userType, signInState) {
+    const identifyObj = new Identify();
+    const formattedUserId = this.formatUserId(userId);
+    // Dont set user id pending privacy conversations
+    // setUserId(formattedUserId);
+    identifyObj.set('userType', userType);
+    identifyObj.set('signInState', signInState);
+
+    if (!this.shouldPutRecord(ALWAYS_PUT)) {
+      this.log(
+        `User properties: userId: ${formattedUserId}, userType: ${userType}, signInState: ${signInState}`
+      );
+    }
+    identify(identifyObj);
+  }
+
+  sendEvent(eventName, payload) {
+    if (this.shouldPutRecord(ALWAYS_PUT)) {
+      track(eventName, payload);
+    } else {
+      this.log(`${eventName}. Payload: ${JSON.stringify({payload})}`);
+    }
+  }
+
+  log(message) {
+    console.log(`[AMPLITUDE ANALYTICS EVENT]: ${message}`);
+  }
+
+  formatUserId(userId) {
+    const userIdString = userId.toString() || 'none';
+    if (!userId) {
+      return userIdString;
+    }
+    if (this.isProductionEnvironment()) {
+      return userIdString.padStart(5, '0');
+    } else {
+      const environment = this.getEnvironment();
+      return `${environment}-${userIdString}`;
+    }
+  }
+
+  /**
```

Comment:
I copied a lot of the logic from here down from [firehose.js](https://github.com/code-dot-org/code-dot-org/blob/26a1aae2322b70e196bf6dfaa0dc04163883a57d/apps/src/lib/util/firehose.js#L64).